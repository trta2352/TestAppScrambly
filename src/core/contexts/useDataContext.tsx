import AsyncStorage from '@react-native-async-storage/async-storage';
import {
  createContext,
  ReactElement,
  useContext,
  useEffect,
  useState,
  ReactNode,
} from 'react';
import { Post } from '../models/Post';
import { DB_POSTS } from '../../utils/constants/DBConstants';
import NetworkService from '../services/NetworkService';

type DataContextType = {
  isLoading: boolean;
  posts: Post[];
  savePosts: (posts: Post[]) => void;
  appError: string | null;
  setAppError: (error: string) => void;
  loadPosts: (override: boolean) => void;
};

const DataContext = createContext<DataContextType>({
  isLoading: false,
  posts: [],
  savePosts: () => {},
  appError: null,
  setAppError: () => {},
  loadPosts: () => {},
});

export const useDataContext = (): DataContextType => useContext(DataContext);

const DataProvider = ({ children }: { children: ReactNode }): ReactElement => {
  const [posts, setPosts] = useState<Post[]>([]);
  const [isLoading, setIsLoading] = useState(false);
  const [appError, setAppError] = useState<string | null>(null);

  const savePosts = async (newPosts: Post[]) => {
    try {
      await AsyncStorage.setItem(DB_POSTS, JSON.stringify(newPosts));
    } catch (error) {
      console.error('Failed to save posts:', error);
    }
  };

  const loadPosts = async (override: boolean) => {
    try {
      setIsLoading(true);
      const postsString = await AsyncStorage.getItem(DB_POSTS);

      if (override || !postsString) {
        const posts = await NetworkService.shared.getPosts();
        setPosts(posts);
        savePosts(posts);
      } else {
        setPosts(JSON.parse(postsString));
      }
      setIsLoading(false);
    } catch (error) {
      console.error('Failed to load posts:', error);
      setIsLoading(false);
      if (override) {
        setAppError('Failed to load posts');
      }
    }
  };

  useEffect(() => {
    loadPosts(false);
  }, []);

  return (
    <DataContext.Provider
      value={{
        posts,
        savePosts: savePosts,
        isLoading,
        appError,
        setAppError,
        loadPosts,
      }}>
      {children}
    </DataContext.Provider>
  );
};

export default DataProvider;
