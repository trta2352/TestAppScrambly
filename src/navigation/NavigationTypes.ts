export type RootStackParamList = {
  SplashScreen: undefined;
  HomeScreen: undefined;
};
