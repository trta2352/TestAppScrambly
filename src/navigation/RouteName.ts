export enum RouteName {
  SplashScreen = 'SplashScreen',
  HomeScreen = 'HomeScreen',
}
