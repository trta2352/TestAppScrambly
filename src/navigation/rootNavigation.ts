import {
  StackActions,
  createNavigationContainerRef,
} from '@react-navigation/native';

export const navigationRef = createNavigationContainerRef();

export function navigate(name, params) {
  if (navigationRef.isReady()) {
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    navigationRef.navigate(name, params);
  }
}

export function replace(name, params) {
  if (navigationRef.isReady()) {
    navigationRef.dispatch(StackActions.replace(name, params));
  }
}

export function goBack() {
  if (navigationRef.isReady()) {
    navigationRef.goBack();
  }
}

export function reset(name, params) {
  if (navigationRef.isReady()) {
    navigationRef.reset({
      index: 0,
      routes: [{ name: name, params: params }],
    });
  }
}
