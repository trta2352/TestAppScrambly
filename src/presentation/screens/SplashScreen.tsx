import React, { useEffect } from 'react';
import { View, Text, StyleSheet, ActivityIndicator } from 'react-native';
import { RouteName } from '../../navigation/RouteName';
import { useStackNavigation } from '../../navigation/navigation';

const SplashScreen: React.FC = () => {
  const navigation = useStackNavigation();
  useEffect(() => {
    const timer = setTimeout(() => {
      // navigation.navigate(RouteName.HomeScreen);
    }, 3000);

    return () => clearTimeout(timer);
  }, []);

  return (
    <View style={styles.container}>
      <ActivityIndicator size="large" color="#007AFF" />
      <Text style={styles.text}>Loading...</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#fff',
  },
  text: {
    marginTop: 20,
    fontSize: 18,
    color: '#333',
  },
});

export default SplashScreen;
