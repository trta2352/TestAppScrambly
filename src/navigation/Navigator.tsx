import { NavigationContainer } from '@react-navigation/native';

import SplashScreen from '../presentation/screens/SplashScreen';
import HomeScreen from '../presentation/screens/HomeScreen';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { RouteName } from './RouteName';
import { RootStackParamList } from './NavigationTypes';
import { navigationRef } from './rootNavigation';

const Stack = createNativeStackNavigator<RootStackParamList, 'RootStack'>();

function RootStack() {
  return (
    <Stack.Navigator
      initialRouteName={RouteName.SplashScreen}
      id="RootStack"
      screenOptions={{ headerShown: false }}>
      <Stack.Screen name={RouteName.SplashScreen} component={SplashScreen} />
      <Stack.Screen name={RouteName.HomeScreen} component={HomeScreen} />
    </Stack.Navigator>
  );
}

export default function Navigator() {
  return (
    <NavigationContainer ref={navigationRef}>
      <RootStack />
    </NavigationContainer>
  );
}
