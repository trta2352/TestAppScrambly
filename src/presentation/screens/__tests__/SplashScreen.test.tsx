import React from 'react';
import { render, act } from '@testing-library/react-native';
import SplashScreen from '../SplashScreen'; // Adjust the path as needed
import * as DataContext from '../../../core/contexts/useDataContext';
import { useStackNavigation } from '../../../navigation/navigation';
import { RouteName } from '../../../navigation/RouteName';

jest.mock('lottie-react-native', () => 'LottieView');
jest.mock('../../../core/contexts/useDataContext');
jest.mock('../../../navigation/navigation');

describe('SplashScreen', () => {
  const mockNavigate = jest.fn();

  beforeEach(() => {
    (useStackNavigation as jest.Mock).mockReturnValue({
      navigate: mockNavigate,
    });
    (DataContext.useDataContext as jest.Mock).mockReturnValue({
      isLoading: false,
    });
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.clearAllMocks();
    jest.useRealTimers();
  });

  it('matches the snapshot', () => {
    const { toJSON } = render(<SplashScreen />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders the Lottie animation', () => {
    const { getByTestId } = render(<SplashScreen />);
    expect(getByTestId('LottieView')).toBeTruthy();
  });

  it('navigates to HomeScreen after 3 seconds', () => {
    render(<SplashScreen />);
    act(() => {
      jest.advanceTimersByTime(3000);
    });
    expect(mockNavigate).toHaveBeenCalledWith(RouteName.HomeScreen);
  });
});
