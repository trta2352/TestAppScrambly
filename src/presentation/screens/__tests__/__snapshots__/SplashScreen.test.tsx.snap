// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashScreen matches the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#f3f6f8",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <LottieView
    autoPlay={true}
    loop={true}
    source={
      {
        "assets": [],
        "ddd": 0,
        "fr": 24,
        "h": 160,
        "ip": 0,
        "layers": [
          {
            "ao": 0,
            "bm": 0,
            "ddd": 0,
            "ind": 0,
            "ip": 0,
            "ks": {
              "a": {
                "k": [
                  -76.375,
                  -2.875,
                  0,
                ],
              },
              "o": {
                "k": [
                  {
                    "e": [
                      30,
                    ],
                    "i": {
                      "x": [
                        0.833,
                      ],
                      "y": [
                        0.833,
                      ],
                    },
                    "n": [
                      "0p833_0p833_0p333_0",
                    ],
                    "o": {
                      "x": [
                        0.333,
                      ],
                      "y": [
                        0,
                      ],
                    },
                    "s": [
                      100,
                    ],
                    "t": 8,
                  },
                  {
                    "e": [
                      100,
                    ],
                    "i": {
                      "x": [
                        0.833,
                      ],
                      "y": [
                        0.833,
                      ],
                    },
                    "n": [
                      "0p833_0p833_0p333_0",
                    ],
                    "o": {
                      "x": [
                        0.333,
                      ],
                      "y": [
                        0,
                      ],
                    },
                    "s": [
                      30,
                    ],
                    "t": 24,
                  },
                  {
                    "t": 40,
                  },
                ],
              },
              "p": {
                "k": [
                  187.875,
                  77.125,
                  0,
                ],
              },
              "r": {
                "k": 0,
              },
              "s": {
                "k": [
                  {
                    "e": [
                      200,
                      200,
                      100,
                    ],
                    "i": {
                      "x": [
                        0.833,
                        0.833,
                        0.833,
                      ],
                      "y": [
                        0.833,
                        0.833,
                        0.833,
                      ],
                    },
                    "n": [
                      "0p833_0p833_0p333_0",
                      "0p833_0p833_0p333_0",
                      "0p833_0p833_0p333_0p333",
                    ],
                    "o": {
                      "x": [
                        0.333,
                        0.333,
                        0.333,
                      ],
                      "y": [
                        0,
                        0,
                        0.333,
                      ],
                    },
                    "s": [
                      100,
                      100,
                      100,
                    ],
                    "t": 8,
                  },
                  {
                    "e": [
                      100,
                      100,
                      100,
                    ],
                    "i": {
                      "x": [
                        0.833,
                        0.833,
                        0.833,
                      ],
                      "y": [
                        0.833,
                        0.833,
                        0.833,
                      ],
                    },
                    "n": [
                      "0p833_0p833_0p333_0",
                      "0p833_0p833_0p333_0",
                      "0p833_0p833_0p333_0p333",
                    ],
                    "o": {
                      "x": [
                        0.333,
                        0.333,
                        0.333,
                      ],
                      "y": [
                        0,
                        0,
                        0.333,
                      ],
                    },
                    "s": [
                      200,
                      200,
                      100,
                    ],
                    "t": 24,
                  },
                  {
                    "t": 40,
                  },
                ],
              },
            },
            "nm": "å½¢ç¶å¾å± 5",
            "op": 40,
            "shapes": [
              {
                "it": [
                  {
                    "d": 1,
                    "mn": "ADBE Vector Shape - Ellipse",
                    "nm": "æ¤­åè·¯å¾ 1",
                    "p": {
                      "k": [
                        0,
                        0,
                      ],
                    },
                    "s": {
                      "k": [
                        18,
                        18,
                      ],
                    },
                    "ty": "el",
                  },
                  {
                    "c": {
                      "k": [
                        1,
                        1,
                        1,
                        1,
                      ],
                    },
                    "lc": 1,
                    "lj": 1,
                    "ml": 4,
                    "mn": "ADBE Vector Graphic - Stroke",
                    "nm": "æè¾¹ 1",
                    "o": {
                      "k": 100,
                    },
                    "ty": "st",
                    "w": {
                      "k": 0,
                    },
                  },
                  {
                    "c": {
                      "k": [
                        0.87,
                        0.42,
                        0.56,
                        1,
                      ],
                    },
                    "mn": "ADBE Vector Graphic - Fill",
                    "nm": "å¡«å 1",
                    "o": {
                      "k": 100,
                    },
                    "ty": "fl",
                  },
                  {
                    "a": {
                      "ix": 1,
                      "k": [
                        0,
                        0,
                      ],
                    },
                    "nm": "åæ¢",
                    "o": {
                      "ix": 7,
                      "k": 100,
                    },
                    "p": {
                      "ix": 2,
                      "k": [
                        -76.482,
                        -3.482,
                      ],
                    },
                    "r": {
                      "ix": 6,
                      "k": 0,
                    },
                    "s": {
                      "ix": 3,
                      "k": [
                        100,
                        100,
                      ],
                    },
                    "sa": {
                      "ix": 5,
                      "k": 0,
                    },
                    "sk": {
                      "ix": 4,
                      "k": 0,
                    },
                    "ty": "tr",
                  },
                ],
                "mn": "ADBE Vector Group",
                "nm": "æ¤­å 1",
                "np": 3,
                "ty": "gr",
              },
            ],
            "sr": 1,
            "st": 0,
            "ty": 4,
          },
          {
            "ao": 0,
            "bm": 0,
            "ddd": 0,
            "ind": 1,
            "ip": 0,
            "ks": {
              "a": {
                "k": [
                  -76.375,
                  -2.875,
                  0,
                ],
              },
              "o": {
                "k": [
                  {
                    "e": [
                      30,
                    ],
                    "i": {
                      "x": [
                        0.833,
                      ],
                      "y": [
                        0.833,
                      ],
                    },
                    "n": [
                      "0p833_0p833_0p333_0",
                    ],
                    "o": {
                      "x": [
                        0.333,
                      ],
                      "y": [
                        0,
                      ],
                    },
                    "s": [
                      100,
                    ],
                    "t": 6,
                  },
                  {
                    "e": [
                      100,
                    ],
                    "i": {
                      "x": [
                        0.833,
                      ],
                      "y": [
                        0.833,
                      ],
                    },
                    "n": [
                      "0p833_0p833_0p333_0",
                    ],
                    "o": {
                      "x": [
                        0.333,
                      ],
                      "y": [
                        0,
                      ],
                    },
                    "s": [
                      30,
                    ],
                    "t": 22,
                  },
                  {
                    "t": 36,
                  },
                ],
              },
              "p": {
                "k": [
                  162.125,
                  76.625,
                  0,
                ],
              },
              "r": {
                "k": 0,
              },
              "s": {
                "k": [
                  {
                    "e": [
                      200,
                      200,
                      100,
                    ],
                    "i": {
                      "x": [
                        0.833,
                        0.833,
                        0.833,
                      ],
                      "y": [
                        0.833,
                        0.833,
                        0.833,
                      ],
                    },
                    "n": [
                      "0p833_0p833_0p333_0",
                      "0p833_0p833_0p333_0",
                      "0p833_0p833_0p333_0p333",
                    ],
                    "o": {
                      "x": [
                        0.333,
                        0.333,
                        0.333,
                      ],
                      "y": [
                        0,
                        0,
                        0.333,
                      ],
                    },
                    "s": [
                      100,
                      100,
                      100,
                    ],
                    "t": 6,
                  },
                  {
                    "e": [
                      100,
                      100,
                      100,
                    ],
                    "i": {
                      "x": [
                        0.833,
                        0.833,
                        0.833,
                      ],
                      "y": [
                        0.833,
                        0.833,
                        0.833,
                      ],
                    },
                    "n": [
                      "0p833_0p833_0p333_0",
                      "0p833_0p833_0p333_0",
                      "0p833_0p833_0p333_0p333",
                    ],
                    "o": {
                      "x": [
                        0.333,
                        0.333,
                        0.333,
                      ],
                      "y": [
                        0,
                        0,
                        0.333,
                      ],
                    },
                    "s": [
                      200,
                      200,
                      100,
                    ],
                    "t": 22,
                  },
                  {
                    "t": 36,
                  },
                ],
              },
            },
            "nm": "å½¢ç¶å¾å± 4",
            "op": 40,
            "shapes": [
              {
                "it": [
                  {
                    "d": 1,
                    "mn": "ADBE Vector Shape - Ellipse",
                    "nm": "æ¤­åè·¯å¾ 1",
                    "p": {
                      "k": [
                        0,
                        0,
                      ],
                    },
                    "s": {
                      "k": [
                        18,
                        18,
                      ],
                    },
                    "ty": "el",
                  },
                  {
                    "c": {
                      "k": [
                        1,
                        1,
                        1,
                        1,
                      ],
                    },
                    "lc": 1,
                    "lj": 1,
                    "ml": 4,
                    "mn": "ADBE Vector Graphic - Stroke",
                    "nm": "æè¾¹ 1",
                    "o": {
                      "k": 100,
                    },
                    "ty": "st",
                    "w": {
                      "k": 0,
                    },
                  },
                  {
                    "c": {
                      "k": [
                        0.81,
                        0.55,
                        0.82,
                        1,
                      ],
                    },
                    "mn": "ADBE Vector Graphic - Fill",
                    "nm": "å¡«å 1",
                    "o": {
                      "k": 100,
                    },
                    "ty": "fl",
                  },
                  {
                    "a": {
                      "ix": 1,
                      "k": [
                        0,
                        0,
                      ],
                    },
                    "nm": "åæ¢",
                    "o": {
                      "ix": 7,
                      "k": 100,
                    },
                    "p": {
                      "ix": 2,
                      "k": [
                        -76.482,
                        -3.482,
                      ],
                    },
                    "r": {
                      "ix": 6,
                      "k": 0,
                    },
                    "s": {
                      "ix": 3,
                      "k": [
                        100,
                        100,
                      ],
                    },
                    "sa": {
                      "ix": 5,
                      "k": 0,
                    },
                    "sk": {
                      "ix": 4,
                      "k": 0,
                    },
                    "ty": "tr",
                  },
                ],
                "mn": "ADBE Vector Group",
                "nm": "æ¤­å 1",
                "np": 3,
                "ty": "gr",
              },
            ],
            "sr": 1,
            "st": 0,
            "ty": 4,
          },
          {
            "ao": 0,
            "bm": 0,
            "ddd": 0,
            "ind": 2,
            "ip": 0,
            "ks": {
              "a": {
                "k": [
                  -76.375,
                  -2.875,
                  0,
                ],
              },
              "o": {
                "k": [
                  {
                    "e": [
                      30,
                    ],
                    "i": {
                      "x": [
                        0.833,
                      ],
                      "y": [
                        0.833,
                      ],
                    },
                    "n": [
                      "0p833_0p833_0p333_0",
                    ],
                    "o": {
                      "x": [
                        0.333,
                      ],
                      "y": [
                        0,
                      ],
                    },
                    "s": [
                      100,
                    ],
                    "t": 4,
                  },
                  {
                    "e": [
                      100,
                    ],
                    "i": {
                      "x": [
                        0.833,
                      ],
                      "y": [
                        0.833,
                      ],
                    },
                    "n": [
                      "0p833_0p833_0p333_0",
                    ],
                    "o": {
                      "x": [
                        0.333,
                      ],
                      "y": [
                        0,
                      ],
                    },
                    "s": [
                      30,
                    ],
                    "t": 20,
                  },
                  {
                    "t": 32,
                  },
                ],
              },
              "p": {
                "k": [
                  135.625,
                  76.625,
                  0,
                ],
              },
              "r": {
                "k": 0,
              },
              "s": {
                "k": [
                  {
                    "e": [
                      200,
                      200,
                      100,
                    ],
                    "i": {
                      "x": [
                        0.833,
                        0.833,
                        0.833,
                      ],
                      "y": [
                        0.833,
                        0.833,
                        0.833,
                      ],
                    },
                    "n": [
                      "0p833_0p833_0p333_0",
                      "0p833_0p833_0p333_0",
                      "0p833_0p833_0p333_0p333",
                    ],
                    "o": {
                      "x": [
                        0.333,
                        0.333,
                        0.333,
                      ],
                      "y": [
                        0,
                        0,
                        0.333,
                      ],
                    },
                    "s": [
                      100,
                      100,
                      100,
                    ],
                    "t": 4,
                  },
                  {
                    "e": [
                      100,
                      100,
                      100,
                    ],
                    "i": {
                      "x": [
                        0.833,
                        0.833,
                        0.833,
                      ],
                      "y": [
                        0.833,
                        0.833,
                        0.833,
                      ],
                    },
                    "n": [
                      "0p833_0p833_0p333_0",
                      "0p833_0p833_0p333_0",
                      "0p833_0p833_0p333_0p333",
                    ],
                    "o": {
                      "x": [
                        0.333,
                        0.333,
                        0.333,
                      ],
                      "y": [
                        0,
                        0,
                        0.333,
                      ],
                    },
                    "s": [
                      200,
                      200,
                      100,
                    ],
                    "t": 20,
                  },
                  {
                    "t": 32,
                  },
                ],
              },
            },
            "nm": "å½¢ç¶å¾å± 3",
            "op": 40,
            "shapes": [
              {
                "it": [
                  {
                    "d": 1,
                    "mn": "ADBE Vector Shape - Ellipse",
                    "nm": "æ¤­åè·¯å¾ 1",
                    "p": {
                      "k": [
                        0,
                        0,
                      ],
                    },
                    "s": {
                      "k": [
                        18,
                        18,
                      ],
                    },
                    "ty": "el",
                  },
                  {
                    "c": {
                      "k": [
                        1,
                        1,
                        1,
                        1,
                      ],
                    },
                    "lc": 1,
                    "lj": 1,
                    "ml": 4,
                    "mn": "ADBE Vector Graphic - Stroke",
                    "nm": "æè¾¹ 1",
                    "o": {
                      "k": 100,
                    },
                    "ty": "st",
                    "w": {
                      "k": 0,
                    },
                  },
                  {
                    "c": {
                      "k": [
                        0.47,
                        0.31,
                        0.62,
                        1,
                      ],
                    },
                    "mn": "ADBE Vector Graphic - Fill",
                    "nm": "å¡«å 1",
                    "o": {
                      "k": 100,
                    },
                    "ty": "fl",
                  },
                  {
                    "a": {
                      "ix": 1,
                      "k": [
                        0,
                        0,
                      ],
                    },
                    "nm": "åæ¢",
                    "o": {
                      "ix": 7,
                      "k": 100,
                    },
                    "p": {
                      "ix": 2,
                      "k": [
                        -76.482,
                        -3.482,
                      ],
                    },
                    "r": {
                      "ix": 6,
                      "k": 0,
                    },
                    "s": {
                      "ix": 3,
                      "k": [
                        100,
                        100,
                      ],
                    },
                    "sa": {
                      "ix": 5,
                      "k": 0,
                    },
                    "sk": {
                      "ix": 4,
                      "k": 0,
                    },
                    "ty": "tr",
                  },
                ],
                "mn": "ADBE Vector Group",
                "nm": "æ¤­å 1",
                "np": 3,
                "ty": "gr",
              },
            ],
            "sr": 1,
            "st": 0,
            "ty": 4,
          },
          {
            "ao": 0,
            "bm": 0,
            "ddd": 0,
            "ind": 3,
            "ip": 0,
            "ks": {
              "a": {
                "k": [
                  -76.625,
                  -3.125,
                  0,
                ],
              },
              "o": {
                "k": [
                  {
                    "e": [
                      30,
                    ],
                    "i": {
                      "x": [
                        0.833,
                      ],
                      "y": [
                        0.833,
                      ],
                    },
                    "n": [
                      "0p833_0p833_0p333_0",
                    ],
                    "o": {
                      "x": [
                        0.333,
                      ],
                      "y": [
                        0,
                      ],
                    },
                    "s": [
                      100,
                    ],
                    "t": 2,
                  },
                  {
                    "e": [
                      100,
                    ],
                    "i": {
                      "x": [
                        0.833,
                      ],
                      "y": [
                        0.833,
                      ],
                    },
                    "n": [
                      "0p833_0p833_0p333_0",
                    ],
                    "o": {
                      "x": [
                        0.333,
                      ],
                      "y": [
                        0,
                      ],
                    },
                    "s": [
                      30,
                    ],
                    "t": 16,
                  },
                  {
                    "t": 28,
                  },
                ],
              },
              "p": {
                "k": [
                  109.375,
                  76.625,
                  0,
                ],
              },
              "r": {
                "k": 0,
              },
              "s": {
                "k": [
                  {
                    "e": [
                      200,
                      200,
                      100,
                    ],
                    "i": {
                      "x": [
                        0.833,
                        0.833,
                        0.833,
                      ],
                      "y": [
                        0.833,
                        0.833,
                        0.833,
                      ],
                    },
                    "n": [
                      "0p833_0p833_0p333_0",
                      "0p833_0p833_0p333_0",
                      "0p833_0p833_0p333_0p333",
                    ],
                    "o": {
                      "x": [
                        0.333,
                        0.333,
                        0.333,
                      ],
                      "y": [
                        0,
                        0,
                        0.333,
                      ],
                    },
                    "s": [
                      100,
                      100,
                      100,
                    ],
                    "t": 2,
                  },
                  {
                    "e": [
                      100,
                      100,
                      100,
                    ],
                    "i": {
                      "x": [
                        0.833,
                        0.833,
                        0.833,
                      ],
                      "y": [
                        0.833,
                        0.833,
                        0.833,
                      ],
                    },
                    "n": [
                      "0p833_0p833_0p333_0",
                      "0p833_0p833_0p333_0",
                      "0p833_0p833_0p333_0p333",
                    ],
                    "o": {
                      "x": [
                        0.333,
                        0.333,
                        0.333,
                      ],
                      "y": [
                        0,
                        0,
                        0.333,
                      ],
                    },
                    "s": [
                      200,
                      200,
                      100,
                    ],
                    "t": 16,
                  },
                  {
                    "t": 28,
                  },
                ],
              },
            },
            "nm": "å½¢ç¶å¾å± 2",
            "op": 40,
            "shapes": [
              {
                "it": [
                  {
                    "d": 1,
                    "mn": "ADBE Vector Shape - Ellipse",
                    "nm": "æ¤­åè·¯å¾ 1",
                    "p": {
                      "k": [
                        0,
                        0,
                      ],
                    },
                    "s": {
                      "k": [
                        18,
                        18,
                      ],
                    },
                    "ty": "el",
                  },
                  {
                    "c": {
                      "k": [
                        1,
                        1,
                        1,
                        1,
                      ],
                    },
                    "lc": 1,
                    "lj": 1,
                    "ml": 4,
                    "mn": "ADBE Vector Graphic - Stroke",
                    "nm": "æè¾¹ 1",
                    "o": {
                      "k": 100,
                    },
                    "ty": "st",
                    "w": {
                      "k": 0,
                    },
                  },
                  {
                    "c": {
                      "k": [
                        0.54,
                        0.81,
                        0.89,
                        1,
                      ],
                    },
                    "mn": "ADBE Vector Graphic - Fill",
                    "nm": "å¡«å 1",
                    "o": {
                      "k": 100,
                    },
                    "ty": "fl",
                  },
                  {
                    "a": {
                      "ix": 1,
                      "k": [
                        0,
                        0,
                      ],
                    },
                    "nm": "åæ¢",
                    "o": {
                      "ix": 7,
                      "k": 100,
                    },
                    "p": {
                      "ix": 2,
                      "k": [
                        -76.482,
                        -3.482,
                      ],
                    },
                    "r": {
                      "ix": 6,
                      "k": 0,
                    },
                    "s": {
                      "ix": 3,
                      "k": [
                        100,
                        100,
                      ],
                    },
                    "sa": {
                      "ix": 5,
                      "k": 0,
                    },
                    "sk": {
                      "ix": 4,
                      "k": 0,
                    },
                    "ty": "tr",
                  },
                ],
                "mn": "ADBE Vector Group",
                "nm": "æ¤­å 1",
                "np": 3,
                "ty": "gr",
              },
            ],
            "sr": 1,
            "st": 0,
            "ty": 4,
          },
          {
            "ao": 0,
            "bm": 0,
            "ddd": 0,
            "ind": 4,
            "ip": 0,
            "ks": {
              "a": {
                "k": [
                  -76.625,
                  -3.375,
                  0,
                ],
              },
              "o": {
                "k": [
                  {
                    "e": [
                      30,
                    ],
                    "i": {
                      "x": [
                        0.833,
                      ],
                      "y": [
                        0.833,
                      ],
                    },
                    "n": [
                      "0p833_0p833_0p333_0",
                    ],
                    "o": {
                      "x": [
                        0.333,
                      ],
                      "y": [
                        0,
                      ],
                    },
                    "s": [
                      100,
                    ],
                    "t": 0,
                  },
                  {
                    "e": [
                      100,
                    ],
                    "i": {
                      "x": [
                        0.833,
                      ],
                      "y": [
                        0.833,
                      ],
                    },
                    "n": [
                      "0p833_0p833_0p333_0",
                    ],
                    "o": {
                      "x": [
                        0.333,
                      ],
                      "y": [
                        0,
                      ],
                    },
                    "s": [
                      30,
                    ],
                    "t": 12,
                  },
                  {
                    "t": 24,
                  },
                ],
              },
              "p": {
                "k": [
                  82.625,
                  76.625,
                  0,
                ],
              },
              "r": {
                "k": 0,
              },
              "s": {
                "k": [
                  {
                    "e": [
                      200,
                      200,
                      100,
                    ],
                    "i": {
                      "x": [
                        0.833,
                        0.833,
                        0.833,
                      ],
                      "y": [
                        0.833,
                        0.833,
                        0.833,
                      ],
                    },
                    "n": [
                      "0p833_0p833_0p333_0",
                      "0p833_0p833_0p333_0",
                      "0p833_0p833_0p333_0p333",
                    ],
                    "o": {
                      "x": [
                        0.333,
                        0.333,
                        0.333,
                      ],
                      "y": [
                        0,
                        0,
                        0.333,
                      ],
                    },
                    "s": [
                      100,
                      100,
                      100,
                    ],
                    "t": 0,
                  },
                  {
                    "e": [
                      100,
                      100,
                      100,
                    ],
                    "i": {
                      "x": [
                        0.833,
                        0.833,
                        0.833,
                      ],
                      "y": [
                        0.833,
                        0.833,
                        0.833,
                      ],
                    },
                    "n": [
                      "0p833_0p833_0p333_0",
                      "0p833_0p833_0p333_0",
                      "0p833_0p833_0p333_0p333",
                    ],
                    "o": {
                      "x": [
                        0.333,
                        0.333,
                        0.333,
                      ],
                      "y": [
                        0,
                        0,
                        0.333,
                      ],
                    },
                    "s": [
                      200,
                      200,
                      100,
                    ],
                    "t": 12,
                  },
                  {
                    "t": 24,
                  },
                ],
              },
            },
            "nm": "å½¢ç¶å¾å± 1",
            "op": 40,
            "shapes": [
              {
                "it": [
                  {
                    "d": 1,
                    "mn": "ADBE Vector Shape - Ellipse",
                    "nm": "æ¤­åè·¯å¾ 1",
                    "p": {
                      "k": [
                        0,
                        0,
                      ],
                    },
                    "s": {
                      "k": [
                        18,
                        18,
                      ],
                    },
                    "ty": "el",
                  },
                  {
                    "c": {
                      "k": [
                        1,
                        1,
                        1,
                        1,
                      ],
                    },
                    "lc": 1,
                    "lj": 1,
                    "ml": 4,
                    "mn": "ADBE Vector Graphic - Stroke",
                    "nm": "æè¾¹ 1",
                    "o": {
                      "k": 100,
                    },
                    "ty": "st",
                    "w": {
                      "k": 0,
                    },
                  },
                  {
                    "c": {
                      "k": [
                        0.34,
                        0.45,
                        0.78,
                        1,
                      ],
                    },
                    "mn": "ADBE Vector Graphic - Fill",
                    "nm": "å¡«å 1",
                    "o": {
                      "k": 100,
                    },
                    "ty": "fl",
                  },
                  {
                    "a": {
                      "ix": 1,
                      "k": [
                        0,
                        0,
                      ],
                    },
                    "nm": "åæ¢",
                    "o": {
                      "ix": 7,
                      "k": 100,
                    },
                    "p": {
                      "ix": 2,
                      "k": [
                        -76.482,
                        -3.482,
                      ],
                    },
                    "r": {
                      "ix": 6,
                      "k": 0,
                    },
                    "s": {
                      "ix": 3,
                      "k": [
                        100,
                        100,
                      ],
                    },
                    "sa": {
                      "ix": 5,
                      "k": 0,
                    },
                    "sk": {
                      "ix": 4,
                      "k": 0,
                    },
                    "ty": "tr",
                  },
                ],
                "mn": "ADBE Vector Group",
                "nm": "æ¤­å 1",
                "np": 3,
                "ty": "gr",
              },
            ],
            "sr": 1,
            "st": 0,
            "ty": 4,
          },
        ],
        "op": 40,
        "v": "4.5.4",
        "w": 280,
      }
    }
    style={
      {
        "height": 200,
        "width": 200,
      }
    }
    testID="LottieView"
  />
</View>
`;
